js/database.js

```diff
@@ -0,0 +1,168 @@
+/*
+ * @constructor
+ */
+
+function Database() {
+	this.db = new Object();
+	this.currDB = new Object();// Necessario porque se não perco a referencia do semestre atual
+
+	function changingSpecialCharacters(word) {
+		return word.toUpperCase()
+			.replace(/[ÀÁÂÃÄÅ]/g, "A")
+			.replace(/Ç/g, "C")
+			.replace(/[ÈÉÊË]/g, "E")
+			.replace(/[ÌÎÍÏ]/g, "I")
+			.replace(/Ð/g, "D")
+			.replace(/Ñ/g, "N")
+			.replace(/[ÒÓÔÕÖØ]/g, "O")
+			.replace(/[ÙÚÛÜ]/g, "U")
+			.replace(/Ý/g, "Y")
+			.replace(/ß/g, "B");
+	}
+
+	function calcScore(haystack, needle, value) {
+		needle.lastIndex = 0;
+		var tmp = haystack.match(needle);
+		if(tmp === null) return 0;
+
+		return tmp.length * value;
+	}
+
+	this.fetchLectureOnDB = function(word) {
+		var self = this;
+		word = changingSpecialCharacters(word);
+
+		var searchWholeString = new Array;
+		var searchPartialString = new Array;
+		word.split(" ").forEach(function(character) {
+				if(character != "") {
+				searchWholeString.push(new RegExp("\\b" + character + "\\b", "g"));
+				searchPartialString.push(new RegExp(character, "g"));
+				}
+				});
+
+		self.result = new Array;
+		for(var code in this.currDB) {
+			var haystack = this.currDB[code];
+			var exactly = false;
+			var score = 0;
+			for(var i = 0; i < searchWholeString.length; i++) {
+				var tmpScore = 0;
+				searchWholeString[i].lastIndex = 0;
+				if(searchWholeString[i].test(haystack.code)) {
+					exactly = true;
+					break;
+				}
+
+				var nameWithoutSpecialChar =  changingSpecialCharacters(haystack.name);
+				tmpScore += calcScore(nameWithoutSpecialChar, searchWholeString[i], 100);
+				tmpScore += calcScore(nameWithoutSpecialChar, searchPartialString[i], 10);
+				tmpScore += calcScore(haystack.code, searchPartialString[i], 10);
+
+				if(tmpScore) {
+					score += tmpScore;
+				}	else {
+					score = 0;
+					break;
+				}
+			}
+
+			if(exactly) {
+				self.result = [haystack];
+				break;
+			}
+			if(score) {
+				haystack.score = score;
+				self.result.score = score;
+				self.result.push(haystack);
+			}
+		}
+
+		self.result.sort(function(first, second) {
+				return first.score - second.score;
+		});
+		self.result.forEach(function(t) {
+				delete t.score;
+		});
+	}
+
+	function loadJSON(callback) {
```

Comment:
>loadJSON [](start = 10, length = 8)

I'd move anything that invokes AJAX requests to a separate file. database.js could then depend on a datasource that would retrieve the data. 

This would reduce the code duplication (`new XMLHttpRequest(); overrideMimeType(); onreadystatechange()`) and reduce coupling with the current storage/retrieval model.